hack/check-code-patterns.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eux
+
+out="$( find . -name '*_test.go' -not -path './vendor/*' -print0 | xargs -0 grep -El 'ioutil\.TempDir' )"
```

Comment:
I'll argue this syntax is better since it prints the occurrence + line numbers:

```
grep -EIrn 'ioutil.\TempDir' | grep _test.go | grep -v vendor/
```

but up to you :)

Reply:
Good point, I'll make sure it shows line numbers. However, I prefer the `find` way because it excludes the vendor folder and files we don't need to check.

Reply:
hmm when I replace `grep` with `echo grep` I only get 1 execution and that's with the last result. I recommend debugging this.

Reply:
I fear I don't fully understand your concern.

The latest version will find multiple matches per file across multiple files, including line numbers. Was that your concern?